backend/manager/modules/enginesso/src/main/java/org/ovirt/engine/core/sso/service/TokenCleanupService.java

```diff
@@ -53,6 +53,7 @@ public static void cleanupSsoSession(
         try {
             ssoContext.removeSsoSession(ssoSession.getAccessToken());
             HttpSession existingSession = ssoSession.getHttpSession();
+            String refreshToken = ssoSession.getRefreshToken();
```

Comment:
Maybe you could add a debug message if the refresh token is null or empty to confirm an assumption that it is happening on the OST.

Reply:
it's in keycloak.log and then also propagated back to engine.log. It shows up as parsing error which is imho good enough to distinguish from expired/invalid value